sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -716,6 +741,7 @@ trait HashJoin extends BaseJoinExec with CodegenSupport {
     } else {
       s"$existsVar = true;"
     }
+    val checkCondition = s"$numMatched.add(1);\n$conditionDef"
```

Comment:
ditto. We had better add this before line 712 and line 717 directly.

Reply:
@dongjoon-hyun I am slightly confused about this part. I added `$numMatched.add(1);` within the checkCondition itself. I did this by adding it directly  before line 742 and line 737 directly. That way every time checkCondition is invoked the metric is incremented. I hope that is what you are looking for. If not, could you please elaborate where I am making a mistake.